packages/block-editor/src/components/use-block-drop-zone/index.native.js

```diff
@@ -111,50 +116,82 @@ export default function useBlockDropZone( {
 	rootClientId: targetRootClientId = '',
 } = {} ) {
 	const targetBlockIndex = useSharedValue( null );
+	const dragPosition = {
+		x: useSharedValue( 0 ),
+		y: useSharedValue( 0 ),
+	};
+	const prevDragPosition = {
+		x: useSharedValue( 0 ),
+		y: useSharedValue( 0 ),
+	};
 
 	const { getBlockListSettings, getSettings } = useSelect( blockEditorStore );
 	const { blocksLayouts, getBlockLayoutsOrderedByYCoord } =
 		useBlockListContext();
 
 	const getSortedBlocksLayouts = useCallback( () => {
 		return getBlockLayoutsOrderedByYCoord( blocksLayouts.current );
+		// We use the value of `blocksLayouts` as the dependency.
+		// eslint-disable-next-line react-hooks/exhaustive-deps
 	}, [ blocksLayouts.current ] );
 
 	const isRTL = getSettings().isRTL;
 
 	const onBlockDrop = useOnBlockDrop();
 
-	const throttled = useThrottle(
-		useCallback(
-			( event ) => {
-				const sortedBlockLayouts = getSortedBlocksLayouts();
-
-				const targetIndex = getNearestBlockIndex(
-					sortedBlockLayouts,
-					{ x: event.x, y: event.y },
-					getBlockListSettings( targetRootClientId )?.orientation,
-					isRTL
-				);
-				if ( targetIndex !== null ) {
-					targetBlockIndex.value = targetIndex ?? 0;
-				}
-			},
-			[
-				getSortedBlocksLayouts,
-				getNearestBlockIndex,
-				getBlockListSettings,
-				targetBlockIndex,
-			]
-		),
-		200
+	const updateTargetBlockIndex = useCallback(
+		( event ) => {
+			const sortedBlockLayouts = getSortedBlocksLayouts();
+
+			const targetIndex = getNearestBlockIndex(
+				sortedBlockLayouts,
+				{ x: event.x, y: event.y },
+				getBlockListSettings( targetRootClientId )?.orientation,
+				isRTL
+			);
+			if ( targetIndex !== null ) {
+				targetBlockIndex.value = targetIndex ?? 0;
+			}
+		},
+		[
+			getSortedBlocksLayouts,
+			getBlockListSettings,
+			targetRootClientId,
+			isRTL,
+			targetBlockIndex,
+		]
 	);
 
+	useDerivedValue( () => {
+		const x = dragPosition.x.value;
+		const y = dragPosition.y.value;
+		const prevX = prevDragPosition.x.value;
+		const prevY = prevDragPosition.y.value;
+		// `updateTargetBlockIndex` performs expensive calculations, so we throttle
+		// the call using a offset threshold based on the dragging position.
+		if (
+			Math.abs( x - prevX ) >= UPDATE_TARGET_BLOCK_INDEX_THRESHOLD ||
+			Math.abs( y - prevY ) >= UPDATE_TARGET_BLOCK_INDEX_THRESHOLD
+		) {
+			runOnJS( updateTargetBlockIndex )( { x, y } );
+			prevDragPosition.x.value = x;
+			prevDragPosition.y.value = y;
```

Comment:
From my understanding of [`useDerivedValue`](https://docs.swmansion.com/react-native-reanimated/docs/2.x/api/hooks/useDerivedValue/)'s purpose, ignoring the boolean return value and executing this side effect within `useDerivedValue` feels a bit odd. Is this a common approach? 

I wonder if leveraging the returned boolean in a separate effect Hook is reasonable. I did not test it. I may be complicating things for no reason. 

<details><summary>Effect Hook</summary>

```js
const shouldUpdateTargetBlockIndex = useDerivedValue( () => {
	const x = dragPosition.x.value;
	const y = dragPosition.y.value;
	const prevX = prevDragPosition.x.value;
	const prevY = prevDragPosition.y.value;
	// `updateTargetBlockIndex` performs expensive calculations, so we throttle
	// the call using a offset threshold based on the dragging position.
	if (
		Math.abs( x - prevX ) >= UPDATE_TARGET_BLOCK_INDEX_THRESHOLD ||
		Math.abs( y - prevY ) >= UPDATE_TARGET_BLOCK_INDEX_THRESHOLD
	) {
		return true;
	}
	return false;
} );

useEffect( () => {
	const x = dragPosition.x.value;
	const y = dragPosition.y.value;

	if ( shouldUpdateTargetBlockIndex.value ) {
		runOnJS( updateTargetBlockIndex )( { x, y } );
		prevDragPosition.x.value = x;
		prevDragPosition.y.value = y;
	}
}, [
	dragPosition.x,
	dragPosition.y,
	prevDragPosition.x,
	prevDragPosition.y,
	shouldUpdateTargetBlockIndex,
	updateTargetBlockIndex,
] );
```

</details>

Reply:
> From my understanding of [`useDerivedValue`](https://docs.swmansion.com/react-native-reanimated/docs/2.x/api/hooks/useDerivedValue/)'s purpose, ignoring the boolean return value and executing this side effect within `useDerivedValue` feels a bit odd. Is this a common approach?

I agree it's uncommon to not use the result of the `userDerivedValue` hook, but seems the only approach to listen to changes in shared values ([reference](https://docs.swmansion.com/react-native-reanimated/docs/2.x/fundamentals/shared-values#reactiveness-with-shared-values)). I found [this example](https://docs.swmansion.com/react-native-reanimated/docs/2.x/api/miscellaneous/runOnJS#example) in the Reanimated documentation about running code in the JS thread and the hook.

Reply:
> I wonder if leveraging the returned boolean in a separate effect Hook is reasonable. I did not test it. I may be complicating things for no reason.

Actually, that was my first approach but I couldn't make it work 😅. Looks like changes to `shouldUpdateTargetBlockIndex`, which happen in a worklet, are not triggering the effect hook. That's why I decided to go with using `runOnJS` directly in the `useDerivedValue` hook.